### DISCUSSION TITLE
Auto-reload SMTP certificate

### DISCUSSION THREAD

**Discussion by SonicGD** [postalserver/postal#1468] [Feature suggestions]
Hello. Correct me if i'm wrong, but Postal doesn't monitor SMTP certificate files and if they were changed we need to restart whole server to apply them. Can this be changed? Maybe by using inotify or smth? 

We are using Postal in Kuberentes and certificates are managed via secrets. And we need to update certificates without restarting server.

Thanks.

**Comment by willpower232** [comment]
Are you able to trigger `postal restart` in this setup?

This is an issue for Lets Encrypt based setups too, I typically modify the `/etc/cron.d/certbot` file to include whatever restart command.

**Comment by SonicGD** [comment]
I think we can. Thanks, it's can be used. But still, it restarts everything. Would be much better if it allows specify "worker" to restart.

**Comment by willpower232** [comment]
Potentially but the restart command does smoothly restart the processes so the difference with only restarting workers would be minimal I would believe (unless someone wants to correct me)

**Reply by dtrckd** [reply]
Hi, the restart command does not seem to exist in current version. Is it assumed that we should restart like this `postal stop && postal start` ?

**Reply by willpower232** [reply]
ah interesting, I believe an appropriate command would be `postal dc restart smtp` and maybe `postal dc restart worker` if that is necessary as well

**Reply by dtrckd** [reply]
Ah, very nice ! I didn't  know that shortcut, thank you sir :).
